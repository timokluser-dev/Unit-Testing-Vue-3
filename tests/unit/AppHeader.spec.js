import AppHeader from '@/components/AppHeader'
import { mount } from '@vue/test-utils'
import serializer from 'jest-serializer-vue-tjw'

beforeEach(() => {
  // inline register serializer
  expect.addSnapshotSerializer(serializer)
})

describe('AppHeader', () => {
  test('if user is not logged in, do not show logout button', () => {
    const wrapper = mount(AppHeader)
    expect(wrapper.find('button').isVisible()).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('if logged in, show logout button', async () => {
    const wrapper = mount(AppHeader)
    await wrapper.setData({ loggedIn: true })

    expect(wrapper.find('button').isVisible()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('if user is initially logged out and logs in', () => {
    const wrapper = mount(AppHeader)
    // with snapshot name (when using multiple)
    expect(wrapper.html()).toMatchSnapshot("initial")

    wrapper.setData({ loggedIn: true })

    expect(wrapper.html()).toMatchSnapshot("logged in")
  })
})
