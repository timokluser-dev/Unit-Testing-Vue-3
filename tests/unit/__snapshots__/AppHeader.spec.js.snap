// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader if logged in, show logout button 1`] = `
<div class="app-header">
  <button style>Logout</button>
</div>
`;

exports[`AppHeader if user is initially logged out and logs in: initial 1`] = `
<div class="app-header">
  <button style="display: none;">Logout</button>
</div>
`;

exports[`AppHeader if user is initially logged out and logs in: logged in 1`] = `
<div class="app-header">
  <button style="display: none;">Logout</button>
</div>
`;

exports[`AppHeader if user is not logged in, do not show logout button 1`] = `
<div class="app-header">
  <button style="display: none;">Logout</button>
</div>
`;
